packages/components/inputs/time-input/src/time-input.tsx

```diff
@@ -129,12 +128,21 @@ const hasMilliseconds = (parsedTime: {
   milliseconds: number;
 }) => parsedTime.milliseconds !== 0;
 
+// Calling `eventTarget.dispatchEvent` does not natively work in React.
+// Instead, we need to grab the element value setter, set the value, and dispatch a change event.
+const dispatchReactChangeEvent = (node: HTMLInputElement, value?: string) => {
+  const setValue = Object.getOwnPropertyDescriptor(
+    node.constructor.prototype,
+    'value'
+  )?.set;
+  setValue?.call(node, value);
+  node.dispatchEvent(new Event('change', { bubbles: true }));
+};
+
 const TimeInput = (props: TTimeInputProps) => {
   const id = useFieldId(props.id, sequentialId);
   const intl = useIntl();
   const element = useRef<HTMLInputElement>(null);
-  // Keep internal state to allow clearing the value manually (`onClear`).
-  const [controlledValue, setControlledValue] = useState(props.value);
```

Comment:
This was the other source of the problems. Since we now set the value using our custom `dispatchReactChangeEvent`, we can simply forward the `props.value` as-is.